### PR TITLE
Add a new rule, how to discard specific unstaged changes

### PR DESCRIPTION
**Pull request by codeofcarson**
I ran into a problem where I wanted to get rid of some hunks in my changes but wasn't ready to commit, and I wasn't having an easy time doing it because some of the hunks were in files where I had made changes I wanted to keep.

I could discard all changes to a file with git checkout <file> but needed a way to sift out the good ones and keep them in my working copy.

Stash ended up being the best tool since you can stash -p and stash only specific hunks.

### CODE REVIEW

**Comment by codeofcarson** [comment]
If I was required to open an issue first I can do that, please let me know. I am very new to contributing to non-work projects.

Thanks

**Comment by RichardLitt** [comment]
You're good, @codeofcarson, this was the right way to do this! Thank you. As it happened, I ran across this issue today, and didn't think of resetting sections this way. Good thinking! 

Might be good to mention that what you're doing with `stash` followed by `reset --hard` - effectively, you're suggesting that I save snippets I want in stash, and then hard reset everything else. There's another strategy, too:

```
git stash -p 
# Select all of the snippets you don't want to save
git stash drop
```

This will delete all of the snippets in the latest stash, while keeping old stashes and not effecting lines you don't add to the stash. Perhaps you should mention this strategy, too? And add a `# Select all of the snippets you want to save` in between `stash -p` and `git reset --hard` in your example. 

How does that sound?

**Comment by RichardLitt** [comment]
I see this is your first PR - Well done! ⭐️ You're awesome. In case you don't know, to add the changes I suggested, just add another commit on top of this, or amend your commit and force push to this branch.

**Comment by codeofcarson** [comment]
Thanks!

I was slightly nervous about it but I'm glad I got it mostly correct.

I have made the changes you have described and force pushed an amended version of the commit.

This should incorporate all of your suggestions which, in my opinion, improve the quality of my commit.

File: README.md [hunk 1]
```diff
@@ -279,6 +280,27 @@ To reset only a specific file, you can use that the filename as the argument:
...
+$ git stash pop
+```
+
+Alternate Strategy, stash undesired changes, drop stash
+
+```sh
+$ git stash -p
+# Select all of the snippets you don't want to save
+git stash drop
+
```
Comment (RichardLitt) [inline README.md]: Needs closing ```.

File: README.md [hunk 2]
```diff
@@ -279,6 +280,27 @@ To reset only a specific file, you can use that the filename as the argument:
 $ git reset filename
 ```
 
+<a href="i-want-to-discard-specific-unstaged-changes"></a>
+### I want to discard specific unstaged changes
+
+When you want to get rid of some, but not all changes in your working copy
+
+First strategy, stash all good changes, reset working copy, reapply good changes
```
Comment (RichardLitt) [inline README.md]: Add a period at the end :)

File: README.md [hunk 3]
```diff
@@ -279,6 +280,27 @@ To reset only a specific file, you can use that the filename as the argument:
 $ git reset filename
 ```
 
+<a href="i-want-to-discard-specific-unstaged-changes"></a>
+### I want to discard specific unstaged changes
+
+When you want to get rid of some, but not all changes in your working copy
```
Comment (RichardLitt) [inline README.md]: Add a period at the end.

File: README.md [hunk 4]
```diff
@@ -279,6 +280,27 @@ To reset only a specific file, you can use that the filename as the argument:
...
+First strategy, stash all good changes, reset working copy, reapply good changes
+
+```sh
+$ git stash -p
+# Select all of the snippets you want to save
+$ git reset --hard
+$ git stash pop
+```
+
+Alternate Strategy, stash undesired changes, drop stash
```
Comment (RichardLitt) [inline README.md]: Uncapitalize 'Strategy'

**Comment by codeofcarson** [comment]
Fixed.

**Comment by RichardLitt** [comment]
Awesome! Thank you! <3 🌴

**Comment by codeofcarson** [comment]
Happy to help. Great experience, would PR again 10/10.

**Pull request merged**